pkg/wait/wait_for_ready.go

```diff
@@ -111,20 +111,20 @@ func NoopMessageCallback() MessageCallback {
 func (w *waitForReadyConfig) Wait(watcher watch.Interface, name string, options Options, msgCallback MessageCallback) (error, time.Duration) {
 
 	timeout := options.timeoutWithDefault()
-	floatingTimeout := timeout
+	timeoutTimer := time.NewTimer(timeout)
```

Comment:
Reusing the same timer prevents the timer leak caused by `time.After`. It is also cleaner because the floatingTImeout math is no longer needed.